Tools/clinic/clinic.py

```diff
@@ -2044,23 +2046,30 @@ class Clinic:
 
 """
 
-    def __init__(self, language, printer=None, *, verify=True, filename=None):
+    def __init__(
+            self,
+            language: Language,
+            printer: BlockPrinter | None = None,
+            *,
+            verify: bool = True,
+            filename: str | None = None
+    ) -> None:
         # maps strings to Parser objects.
         # (instantiated from the "parsers" global.)
-        self.parsers = {}
-        self.language = language
+        self.parsers: ParserDict = {}
```

Comment:
I'd prefer for the `ParserDict` definition to be put somewhere above the place where the name is first used in an annotation (line 2059). I think that would improve readability.